BritneyS/DynamicTableView
# Multiple view

## What does this PR do :question:
This PR is a simply demonstrated method of displaying multiple custom reusable views in one tableview conditionally. This PR also demonstrates a different way to hide cells conditionally through performing the deletion and reload operations in `performBatchUpdates`.

The reusable views are `LabelCell` and `ImageCell`, and they are both of type `DynamicTableViewCell`.

## How to test it :microscope:
1. Check out this branch
2. Run the app
3. Note that in the table view the displays the words to the phrase `"The quick brown fox jumps over the lazy dog"`, the cell that would contain the term `lazy` is hidden, and the cell that would contain `dog` is instead replaced by a cell containing an image of a doggo! 🐶


## Any background info you would like to include :white_check_mark:
This PR contains a more dynamic way to hide cells: instead of setting the height of a cell to zero (as in PR https://github.com/BritneyS/DynamicTableView/pull/2), the cell row is deleted. I removed any logic that was related to the cell's `isHidden` property; this seemed to make hard coding height values in `heightForRowAt` necessary, which is not ideal.

I want to explore the possibility of using enums to contain the state of each kind of view.

## Screenshots (if applicable) :camera:
**3-15-19 Updated screenshot to show Autolayout fixes:**
![Simulator Screen Shot - iPhone XR - 2019-03-15 at 10 52 41](https://user-images.githubusercontent.com/8409475/54440140-83ee6b80-4710-11e9-9cb2-1bc81cb8752d.png)
